### DISCUSSION TITLE
Upsert with js SDK

### DISCUSSION THREAD

**Discussion by pr0t0n-cmd** [supabase/supabase#20794] [Questions]
I am having a problem with supabase js sdk. I am trying to use upsert but it says:

ERROR:  {
  code: '42501',
  details: null,
  hint: null,
  message: 'new row violates row-level security policy for table "customers"'
}

I have a page where i can insert records without any problems and if i try update it works fine, what could be the issue? thanks

**Comment by GaryAustin1** [comment]
Not enough info to know.  But clearly you don't meet RLS.
You need to meet policies for Select, Update and Insert to do an Upsert.

**Reply by pr0t0n-cmd** [reply]
Hello, i can do those operations with the same account using the SDK. Select, Update and Insert work fine. what can i check to give more details?

**Reply by GaryAustin1** [reply]
You will have to show your policies and likely upsert data.  You could try to make them all true for any user then slowly add back until it breaks.  Your error is clear though you don't meet it for some reason.  Also make sure if you are testing other calls they are done at the exact same spot in your code so you have the same session conditions.
